### PR TITLE
[Doc] Little Documentation how to use the SDK in the browser

### PR DESCRIPTION
**Pull request by FrankEssenberger**
## Context

I added a little snippet on how to use the SDK on the browser and what are things to consider.

## Definition of Done

- [x] Locally tested the documentation content
- [x] npm run build runs without errors
- [x] result on the server looks fine

### CODE REVIEW

File: docs/js/features/browser/sdk-in-browser.mdx [hunk 1]
```diff
@@ -12,12 +11,134 @@ keywords:
 - sap cloud sdk
 ---
 
-The SAP Cloud SDK for JavaScript can be used both as a backend and frontend library when used in a browser. Because of the specifics of a browser environment, some features might be unavailable. To help you get up and running faster in a browser, we'll outline the main steps and caveats of using SDK on the frontend.
+The SAP Cloud SDK for JavaScript can be used both as a backend and frontend library when used in a browser.
+Because of the specifics of a browser environment, some features might be unavailable.
+To help you get up and running faster in a browser, we'll outline the main steps and caveats of using SDK on the frontend.
+We will use `Angular` to illustrate the usage/
```
Comment (artemkovalyov) [inline docs/js/features/browser/sdk-in-browser.mdx:17]:
```suggestion
We will use `Angular` to illustrate the usage.
```

File: docs/js/features/browser/sdk-in-browser.mdx [hunk 2]
```diff
@@ -12,12 +11,134 @@ keywords:
...
 
-## Configuring destinations
+You have a front-end application set up.
+For example an [Angular application](https://angular.io/cli/new) via `ng new myProject`.
 
-## Making requests
+## Include the SDK
 
-## Integration with frontend frameworks
+As for most examples we will use the business partner API.
```
Comment (artemkovalyov) [inline docs/js/features/browser/sdk-in-browser.mdx:26]:
```suggestion
For most examples we will use the business partner API.
```

File: docs/js/features/browser/sdk-in-browser.mdx [hunk 3]
```diff
@@ -12,12 +11,134 @@ keywords:
...
+
+async function callApi(){
+    const result await BusinessPartner.requestBuilder().getAll().execute(myDest)
+    console.log(result)
+}
+```
+
+:::note
+For the destination lookup the SDK relies on `nodejs` functionality like process.env for example.
+However, the request building, execution and result parsing does not require any `nodejs` modules and can be used in the frontend.
```
Comment (artemkovalyov) [inline docs/js/features/browser/sdk-in-browser.mdx:43]:
```suggestion
For the destination lookup the SDK relies on `Node.js` functionality like `process.env` for example.
However, the request building, execution and result parsing does not require any `Node.js` modules and can be used in the front-end.
```

File: docs/js/features/browser/sdk-in-browser.mdx [hunk 4]
```diff
@@ -12,12 +11,134 @@ keywords:
...
+Hence you have to create the destination variable `myDest` manually.
+:::
+
+When you run `npm run build` your project should compile.
+Currently, we are facing compilation errors of the kind `Module not found: Error: Can't resolve 'https' in ...'`
+Please checkout the [workaround](#workaround) for these kind of errors.
```
Comment (artemkovalyov) [inline docs/js/features/browser/sdk-in-browser.mdx:49]:
```suggestion
When you run `npm run build` your project should compile.
Currently, we are facing compilation errors of the kind `Module not found: Error: Can't resolve 'https' in ...'`
Please check out the [workaround](#workaround) for these kinds of errors.
```

File: docs/js/features/browser/sdk-in-browser.mdx [hunk 5]
```diff
@@ -12,12 +11,134 @@ keywords:
...
+  password: 'myPassowrd'
+}
+```
+
+When you execute `npm run serve` your project is compiled and hosted on a local server.
+
+:::note
+When you trigger a request from a browser to the remote system all modern browser will block this to ensure [same origin policy](https://en.wikipedia.org/wiki/Same-origin_policy).
+One simple way to overcome this issue locally is to disable web security of your  browser temporarily (e.g. --disable-web-security for Chrome)
+Alternatively you can also create a second localhost proxy server which adds the `access-allow-origin-header` to the request.
```
Comment (artemkovalyov) [inline docs/js/features/browser/sdk-in-browser.mdx:70]:
```suggestion
When you trigger a request from a browser to the remote system all modern browsers will block this to ensure [same origin policy](https://en.wikipedia.org/wiki/Same-origin_policy).
One simple way to overcome this issue locally is to disable the web security of your  browser temporarily (e.g. --disable-web-security for Chrome)
Alternatively, you can also create a second localhost proxy server which adds the `access-allow-origin-header` to the request.
```

File: docs/js/features/browser/sdk-in-browser.mdx [hunk 6]
```diff
@@ -12,12 +11,134 @@ keywords:
...
+:::
+
+## Run On Cloud Foundry
+
+For productive usage the methods mentioned above for running locally are of course not needed.
+You simply configure the `xs-app.json` of the used app-router with an entry like this:
```
Comment (artemkovalyov) [inline docs/js/features/browser/sdk-in-browser.mdx:76]:
```suggestion
For productive usage, the methods mentioned above for running locally are of course not needed.
You simply configure the `xs-app.json` of the used app-router with an entry like this:
```

File: docs/js/features/browser/sdk-in-browser.mdx [hunk 7]
```diff
@@ -12,12 +11,134 @@ keywords:
...
+:::note
+This approach does not support principal propagation to On-Premise systems.
+:::
+
+:::note
+With webpack it is possible to [load resources](https://webpack.js.org/plugins/normal-module-replacement-plugin/) for local or a productive usage.
+Using such a switch you can easily adjust the value of the destination for the two cases.
+:::
+
+## <a name="workaround"></a>Workaround
```
Comment (artemkovalyov) [inline docs/js/features/browser/sdk-in-browser.mdx:105]:
```suggestion
## Workaround
```

Comment (artemkovalyov) [reply docs/js/features/browser/sdk-in-browser.mdx:105]: This should be enough for a hashtag link.

File: docs/js/features/browser/sdk-in-browser.mdx [hunk 8]
```diff
@@ -12,12 +11,134 @@ keywords:
...
+
+Currently, we are facing the issue, that on compilation from TS to JS the `nodejs` parts cause problems even if they are not used.
```
Comment (artemkovalyov) [inline docs/js/features/browser/sdk-in-browser.mdx:107]:
```suggestion
Currently, we are facing the issue, that on compilation from TS to JS the `Node.js` parts cause problems even if they are not used.
```

File: sidebars.js [hunk 9]
```diff
@@ -144,6 +144,13 @@ module.exports = {
         type: 'category',
         label: 'Features',
         items: [
+          {
+            type: 'category',
+            label: 'Browser',
+            items:[
+              'js/features/browser/sdk-in-browser'
+            ]
+          },
```
Comment (artemkovalyov) [inline sidebars.js:153]: Do you think we'll have more browser dedicated materials to make it a category?

**Review by artemkovalyov — CHANGES_REQUESTED** [review]
Thanks for creating this. Can't wait to try it out myself.
I left minor suggestions.

File: sidebars.js (hunk 9 above)
Comment (FrankEssenberger) [reply sidebars.js:153]: Hm, actually no. But it was in a browser folder. I would move it to the guides sections?

**Comment by FrankEssenberger** [comment]
With the fixed docusaurus version it works now

**Review by artemkovalyov — APPROVED** [review]
Great job! Merging it.

**Pull request merged**